### DISCUSSION TITLE
Persistent session with xrdp

### DISCUSSION THREAD

**Discussion by saikotek** [neutrinolabs/xrdp#3310] [Q&A] [answered]
The moment you login to xrdp with xorg option selected it automatically spawns new Xorg session. I want to have single session that remote user always connects to, right now I'm using XVnc to connect to local xorg session (to x11vnc session, actually), but I wonder if there is a way to do it without XVnc? Perhaps I can auto-start my own Xorg session and force xrdp to always connect to it?

**Comment by matt335672** [comment]
Auto-connecting is what should happen normally.

Have a look in `/var/log.xrdp-sesman.log` when this happens, and see what it's telling you.

A couple of questions:-
- Have you changed any settings in `/etc/xrdp/sesman.ini`?
- Do you restart `xrdp-sesman.service` on a regular basis? That will cause it.

**Reply by saikotek** [reply]
Hello, thank you for response. I forgot to precise that what I want is to have an active Xorg session BEFORE anyone connects using xrdp. That is because I have a GUI application that needs to run since the system boots up. For this right now I'm using local X11 session and spawn x11vnc to which I connect through xrdp.

**Reply by matt335672** [reply, marked as answer]
You can't connect to the console Xorg session from xrdp, except by using x11vnc.

Have a look at xrdp-sesrun which allows you to start an xrdp session before anyone connects. It's a lot more usable for xrdp v0.10.x

**Reply by saikotek** [reply]
I see. Thanks, I'll check it out.
